security/cmd/istio_ca/main.go

```diff
@@ -396,22 +396,22 @@ func createClientset() *kubernetes.Clientset {
 	return cs
 }
 
-func createCA(core corev1.SecretsGetter) *ca.IstioCA {
+func createCA(client corev1.CoreV1Interface) *ca.IstioCA {
 	var caOpts *ca.IstioCAOptions
 	var err error
 
 	if opts.selfSignedCA {
 		log.Info("Use self-signed certificate as the CA certificate")
 		caOpts, err = ca.NewSelfSignedIstioCAOptions(opts.selfSignedCACertTTL, opts.workloadCertTTL,
 			opts.maxWorkloadCertTTL, opts.trustDomain, opts.dualUse,
-			opts.istioCaStorageNamespace, core)
+			opts.istioCaStorageNamespace, client)
 		if err != nil {
 			fatalf("Failed to create a self-signed Citadel (error: %v)", err)
 		}
 	} else {
 		log.Info("Use certificate from argument as the CA certificate")
 		caOpts, err = ca.NewPluggedCertIstioCAOptions(opts.certChainFile, opts.signingCertFile, opts.signingKeyFile,
-			opts.rootCertFile, opts.workloadCertTTL, opts.maxWorkloadCertTTL)
+			opts.rootCertFile, opts.workloadCertTTL, opts.maxWorkloadCertTTL, opts.istioCaStorageNamespace, client)
 		if err != nil {
 			fatalf("Failed to create an Citadel (error: %v)", err)
 		}
```

Comment:
rather than put ```updateCertInConfigmap``` logic in both ```NewSelfSignedIstioCAOptions ``` and ```NewPluggedCertIstioCAOptions```, how about consolidating in ```createCA``` ?

Reply:
Let's keep them here for now. I'm planning to rewrite this part and move "createCA" into pkg/ directory.